<?xml version="1.0" encoding="UTF-8"?>
<testsuite name="test_room_reservation_tests.TestHotelReservation-20240419201726" tests="6" file="test_room_reservation_tests.py" time="0.093" timestamp="2024-04-19T20:17:26" failures="0" errors="6" skipped="0">
	<testcase classname="test_room_reservation_tests.TestHotelReservation" name="test_duplicated_id_reservation_tests" time="0.005" timestamp="2024-04-19T20:17:26">
		<error type="HotelManagementException" message="JSON Decode Error - Wrong JSON Format"><![CDATA[Traceback (most recent call last):
  File "/Users/ellaz-j/PycharmProjects/G88.2024.T06.EG3/src/main/python/storage/json_store.py", line 37, in load_list_from_file
    room_key_list = json.load(file)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/json/__init__.py", line 293, in load
    return loads(fp.read(),
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/unittest/case.py", line 59, in testPartExecutor
    yield
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/unittest/case.py", line 591, in run
    self._callTestMethod(testMethod)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/unittest/case.py", line 549, in _callTestMethod
    method()
  File "/Users/ellaz-j/PycharmProjects/G88.2024.T06.EG3/src/unittest/python/test_room_reservation_tests.py", line 98, in test_duplicated_id_reservation_tests
    mngr.room_reservation(credit_card=credit_card_number,
  File "/Users/ellaz-j/PycharmProjects/G88.2024.T06.EG3/src/main/python/uc3m_travel/hotel_manager.py", line 96, in room_reservation
    ReservationJsonStore().save_list_to_file()
  File "/Users/ellaz-j/PycharmProjects/G88.2024.T06.EG3/src/main/python/storage/reservation_json_store.py", line 29, in save_list_to_file
    return super().save_list_to_file()
  File "/Users/ellaz-j/PycharmProjects/G88.2024.T06.EG3/src/main/python/storage/json_store.py", line 30, in save_list_to_file
    json.dump(self.load_list_from_file(), file, indent=2)
  File "/Users/ellaz-j/PycharmProjects/G88.2024.T06.EG3/src/main/python/storage/reservation_json_store.py", line 27, in load_list_from_file
    return super().load_list_from_file()
  File "/Users/ellaz-j/PycharmProjects/G88.2024.T06.EG3/src/main/python/storage/json_store.py", line 41, in load_list_from_file
    raise HotelManagementException("JSON Decode Error - Wrong JSON Format") from ex
uc3m_travel.hotel_management_exception.HotelManagementException: JSON Decode Error - Wrong JSON Format
]]></error>
	</testcase>
	<testcase classname="test_room_reservation_tests.TestHotelReservation" name="test_duplicated_reservation_tests" time="0.007" timestamp="2024-04-19T20:17:26">
		<error type="HotelManagementException" message="JSON Decode Error - Wrong JSON Format"><![CDATA[Traceback (most recent call last):
  File "/Users/ellaz-j/PycharmProjects/G88.2024.T06.EG3/src/main/python/storage/json_store.py", line 37, in load_list_from_file
    room_key_list = json.load(file)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/json/__init__.py", line 293, in load
    return loads(fp.read(),
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/unittest/case.py", line 59, in testPartExecutor
    yield
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/unittest/case.py", line 591, in run
    self._callTestMethod(testMethod)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/unittest/case.py", line 549, in _callTestMethod
    method()
  File "/Users/ellaz-j/PycharmProjects/G88.2024.T06.EG3/venv/lib/python3.10/site-packages/freezegun/api.py", line 827, in wrapper
    result = func(*args, **kwargs)
  File "/Users/ellaz-j/PycharmProjects/G88.2024.T06.EG3/src/unittest/python/test_room_reservation_tests.py", line 52, in test_duplicated_reservation_tests
    mngr.room_reservation(credit_card=credit_card_number,
  File "/Users/ellaz-j/PycharmProjects/G88.2024.T06.EG3/src/main/python/uc3m_travel/hotel_manager.py", line 96, in room_reservation
    ReservationJsonStore().save_list_to_file()
  File "/Users/ellaz-j/PycharmProjects/G88.2024.T06.EG3/src/main/python/storage/reservation_json_store.py", line 29, in save_list_to_file
    return super().save_list_to_file()
  File "/Users/ellaz-j/PycharmProjects/G88.2024.T06.EG3/src/main/python/storage/json_store.py", line 30, in save_list_to_file
    json.dump(self.load_list_from_file(), file, indent=2)
  File "/Users/ellaz-j/PycharmProjects/G88.2024.T06.EG3/src/main/python/storage/reservation_json_store.py", line 27, in load_list_from_file
    return super().load_list_from_file()
  File "/Users/ellaz-j/PycharmProjects/G88.2024.T06.EG3/src/main/python/storage/json_store.py", line 41, in load_list_from_file
    raise HotelManagementException("JSON Decode Error - Wrong JSON Format") from ex
uc3m_travel.hotel_management_exception.HotelManagementException: JSON Decode Error - Wrong JSON Format
]]></error>
	</testcase>
	<testcase classname="test_room_reservation_tests.TestHotelReservation" name="test_parametrized_cases_tests [TC1VALID]" time="0.000" timestamp="0001-01-01T00:00:00">
		<error type="HotelManagementException" message="JSON Decode Error - Wrong JSON Format"><![CDATA[Traceback (most recent call last):
  File "/Users/ellaz-j/PycharmProjects/G88.2024.T06.EG3/src/main/python/storage/json_store.py", line 37, in load_list_from_file
    room_key_list = json.load(file)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/json/__init__.py", line 293, in load
    return loads(fp.read(),
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/unittest/case.py", line 59, in testPartExecutor
    yield
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/unittest/case.py", line 498, in subTest
    yield
  File "/Users/ellaz-j/PycharmProjects/G88.2024.T06.EG3/src/unittest/python/test_room_reservation_tests.py", line 163, in test_parametrized_cases_tests
    valor = mngr.room_reservation(credit_card=credit_card_number,
  File "/Users/ellaz-j/PycharmProjects/G88.2024.T06.EG3/src/main/python/uc3m_travel/hotel_manager.py", line 96, in room_reservation
    ReservationJsonStore().save_list_to_file()
  File "/Users/ellaz-j/PycharmProjects/G88.2024.T06.EG3/src/main/python/storage/reservation_json_store.py", line 29, in save_list_to_file
    return super().save_list_to_file()
  File "/Users/ellaz-j/PycharmProjects/G88.2024.T06.EG3/src/main/python/storage/json_store.py", line 30, in save_list_to_file
    json.dump(self.load_list_from_file(), file, indent=2)
  File "/Users/ellaz-j/PycharmProjects/G88.2024.T06.EG3/src/main/python/storage/reservation_json_store.py", line 27, in load_list_from_file
    return super().load_list_from_file()
  File "/Users/ellaz-j/PycharmProjects/G88.2024.T06.EG3/src/main/python/storage/json_store.py", line 41, in load_list_from_file
    raise HotelManagementException("JSON Decode Error - Wrong JSON Format") from ex
uc3m_travel.hotel_management_exception.HotelManagementException: JSON Decode Error - Wrong JSON Format
]]></error>
	</testcase>
	<testcase classname="test_room_reservation_tests.TestHotelReservation" name="test_parametrized_cases_tests [TC2VALID]" time="0.000" timestamp="0001-01-01T00:00:00">
		<error type="HotelManagementException" message="JSON Decode Error - Wrong JSON Format"><![CDATA[Traceback (most recent call last):
  File "/Users/ellaz-j/PycharmProjects/G88.2024.T06.EG3/src/main/python/storage/json_store.py", line 37, in load_list_from_file
    room_key_list = json.load(file)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/json/__init__.py", line 293, in load
    return loads(fp.read(),
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/unittest/case.py", line 59, in testPartExecutor
    yield
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/unittest/case.py", line 498, in subTest
    yield
  File "/Users/ellaz-j/PycharmProjects/G88.2024.T06.EG3/src/unittest/python/test_room_reservation_tests.py", line 163, in test_parametrized_cases_tests
    valor = mngr.room_reservation(credit_card=credit_card_number,
  File "/Users/ellaz-j/PycharmProjects/G88.2024.T06.EG3/src/main/python/uc3m_travel/hotel_manager.py", line 83, in room_reservation
    data_list = ReservationJsonStore().load_list_from_file()
  File "/Users/ellaz-j/PycharmProjects/G88.2024.T06.EG3/src/main/python/storage/reservation_json_store.py", line 27, in load_list_from_file
    return super().load_list_from_file()
  File "/Users/ellaz-j/PycharmProjects/G88.2024.T06.EG3/src/main/python/storage/json_store.py", line 41, in load_list_from_file
    raise HotelManagementException("JSON Decode Error - Wrong JSON Format") from ex
uc3m_travel.hotel_management_exception.HotelManagementException: JSON Decode Error - Wrong JSON Format
]]></error>
	</testcase>
	<testcase classname="test_room_reservation_tests.TestHotelReservation" name="test_parametrized_cases_tests [TC3VALID]" time="0.000" timestamp="0001-01-01T00:00:00">
		<error type="HotelManagementException" message="JSON Decode Error - Wrong JSON Format"><![CDATA[Traceback (most recent call last):
  File "/Users/ellaz-j/PycharmProjects/G88.2024.T06.EG3/src/main/python/storage/json_store.py", line 37, in load_list_from_file
    room_key_list = json.load(file)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/json/__init__.py", line 293, in load
    return loads(fp.read(),
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/unittest/case.py", line 59, in testPartExecutor
    yield
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/unittest/case.py", line 498, in subTest
    yield
  File "/Users/ellaz-j/PycharmProjects/G88.2024.T06.EG3/src/unittest/python/test_room_reservation_tests.py", line 163, in test_parametrized_cases_tests
    valor = mngr.room_reservation(credit_card=credit_card_number,
  File "/Users/ellaz-j/PycharmProjects/G88.2024.T06.EG3/src/main/python/uc3m_travel/hotel_manager.py", line 83, in room_reservation
    data_list = ReservationJsonStore().load_list_from_file()
  File "/Users/ellaz-j/PycharmProjects/G88.2024.T06.EG3/src/main/python/storage/reservation_json_store.py", line 27, in load_list_from_file
    return super().load_list_from_file()
  File "/Users/ellaz-j/PycharmProjects/G88.2024.T06.EG3/src/main/python/storage/json_store.py", line 41, in load_list_from_file
    raise HotelManagementException("JSON Decode Error - Wrong JSON Format") from ex
uc3m_travel.hotel_management_exception.HotelManagementException: JSON Decode Error - Wrong JSON Format
]]></error>
	</testcase>
	<testcase classname="test_room_reservation_tests.TestHotelReservation" name="test_parametrized_cases_tests [TC4VALID]" time="0.081" timestamp="2024-04-19T20:17:26">
		<error type="HotelManagementException" message="JSON Decode Error - Wrong JSON Format"><![CDATA[Traceback (most recent call last):
  File "/Users/ellaz-j/PycharmProjects/G88.2024.T06.EG3/src/main/python/storage/json_store.py", line 37, in load_list_from_file
    room_key_list = json.load(file)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/json/__init__.py", line 293, in load
    return loads(fp.read(),
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/unittest/case.py", line 59, in testPartExecutor
    yield
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/unittest/case.py", line 498, in subTest
    yield
  File "/Users/ellaz-j/PycharmProjects/G88.2024.T06.EG3/src/unittest/python/test_room_reservation_tests.py", line 163, in test_parametrized_cases_tests
    valor = mngr.room_reservation(credit_card=credit_card_number,
  File "/Users/ellaz-j/PycharmProjects/G88.2024.T06.EG3/src/main/python/uc3m_travel/hotel_manager.py", line 83, in room_reservation
    data_list = ReservationJsonStore().load_list_from_file()
  File "/Users/ellaz-j/PycharmProjects/G88.2024.T06.EG3/src/main/python/storage/reservation_json_store.py", line 27, in load_list_from_file
    return super().load_list_from_file()
  File "/Users/ellaz-j/PycharmProjects/G88.2024.T06.EG3/src/main/python/storage/json_store.py", line 41, in load_list_from_file
    raise HotelManagementException("JSON Decode Error - Wrong JSON Format") from ex
uc3m_travel.hotel_management_exception.HotelManagementException: JSON Decode Error - Wrong JSON Format
]]></error>
	</testcase>
</testsuite>
